src/test/java/TransportCommunicationIT.java

```diff
@@ -0,0 +1,155 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opensearch.common.component.Lifecycle;
+import org.opensearch.common.network.NetworkAddress;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.transport.TransportAddress;
+import org.opensearch.test.OpenSearchIntegTestCase;
+import org.opensearch.transport.TransportService;
+import org.opensearch.transport.TransportSettings;
+import java.net.*;
+import java.io.*;
+
+import transportservice.RunPlugin;
+import transportservice.netty4.Netty4Transport;
+import org.opensearch.threadpool.ThreadPool;
+
+public class TransportCommunicationIT extends OpenSearchIntegTestCase {
+
+    private int port;
+    private Settings settings;
+    private final int minPort = 49152;
+    private final int maxPort = 65535;
+    private final String host = "127.0.0.1";
+    private volatile String clientResult;
+
+    @BeforeEach
+    public void setUp() {
+
+        // randomize port selection when creating settings
+        port = getRandomPort();
```

Comment:
No need to create a randomPort here, which is actually within the range 49152-65535. Using an explicit port might not be free sometimes. Let's keep the test port as `0`.